manifests/prepare.pp

```diff
@@ -52,14 +54,103 @@
     }
   }
 
-  # TODO: manage server.cfg contents
-  if $::mcollective_configured {
-    file { $::agent_upgrade::params::mcodirs:
-      ensure => directory,
+  # manage client.cfg and server.cfg contents
+  file { $::agent_upgrade::params::mcodirs:
+    ensure => directory,
+  }
+  # The mco_*_config facts will return the location of mcollective config (or nil), prefering PE over FOSS.
+  if $::mco_server_config {
+    $mco_server = $::agent_upgrade::params::mco_server
+    file { $mco_server:
+      ensure  => file,
+      source  => $::mco_server_config,
+    }
+
+    if $::mco_server_settings {
+      $mco_server_libdir = $::mco_server_settings['libdir']
+      if $mco_server_libdir {
+        $mco_server_libdir_array = split($mco_server_libdir, ':')
+        # Only add the new path if it's not already in libdir
+        if [] == $mco_server_libdir_array.filter |$x| { $x == $::agent_upgrade::params::mco_libdir } {
+          ini_setting { 'server/libdir':
+            section => '',
+            setting => 'libdir',
+            path    => $mco_server,
+            value   => "${mco_server_libdir}:$::agent_upgrade::params::mco_libdir",
+            require => File[$mco_server],
+          }
+        }
+      }
+
+      $mco_server_plugins = $::mco_server_settings['plugin.yaml']
+      if $mco_server_plugins {
+        $mco_server_plugins_array = split($mco_server_plugins, ':')
+        # Only add the new path if it's not already in plugin.yaml
+        if [] == $mco_server_plugins_array.filter |$x| { $x == $::agent_upgrade::params::mco_plugins } {
+          ini_setting { 'server/plugin.yaml':
+            section => '',
+            setting => 'plugin.yaml',
+            path    => $mco_server,
+            value   => "${mco_server_plugins}:$::agent_upgrade::params::mco_plugins",
+            require => File[$mco_server],
+          }
+        }
+      }
+    }
+
+    ini_setting { 'server/logfile':
+      section => '',
+      setting => 'logfile',
+      path    => $mco_server,
+      value   => $::agent_upgrade::params::mco_log,
+      require => File[$mco_server],
     }
-    file { $::agent_upgrade::params::mcoserver:
-      ensure => file,
-      source => $::agent_upgrade::params::oldmcoserver,
+  }
+  if $::mco_client_config {
+    $mco_client = $::agent_upgrade::params::mco_client
+    file { $mco_client:
+      ensure  => file,
+      source  => $::mco_client_config,
+    }
+
+    if $::mco_client_settings {
+      $mco_client_libdir = $::mco_client_settings['libdir']
+      if $mco_client_libdir {
+        $mco_client_libdir_array = split($mco_client_libdir, ':')
```

Comment:
I think we need to take in account windows vs posix when splitting the libdir, as windows agents will use semicolon as the separator.